src/neo/SmartContract/ApplicationEngine.OpCodePrices.cs

```diff
@@ -7,190 +7,190 @@ partial class ApplicationEngine
+            [OpCode.STLOC5] = 1 << 1,
+            [OpCode.STLOC6] = 1 << 1,
+            [OpCode.STLOC] = 1 << 1,
+            [OpCode.LDARG0] = 1 << 1,
+            [OpCode.LDARG1] = 1 << 1,
+            [OpCode.LDARG2] = 1 << 1,
+            [OpCode.LDARG3] = 1 << 1,
+            [OpCode.LDARG4] = 1 << 1,
+            [OpCode.LDARG5] = 1 << 1,
+            [OpCode.LDARG6] = 1 << 1,
+            [OpCode.LDARG] = 1 << 1,
+            [OpCode.STARG0] = 1 << 1,
+            [OpCode.STARG1] = 1 << 1,
+            [OpCode.STARG2] = 1 << 1,
+            [OpCode.STARG3] = 1 << 1,
+            [OpCode.STARG4] = 1 << 1,
+            [OpCode.STARG5] = 1 << 1,
+            [OpCode.STARG6] = 1 << 1,
+            [OpCode.STARG] = 1 << 1,
+            [OpCode.NEWBUFFER] = 1 << 11,
+            [OpCode.MEMCPY] = 1 << 11,
+            [OpCode.CAT] = 1 << 11,
+            [OpCode.SUBSTR] = 1 << 11,
+            [OpCode.LEFT] = 1 << 11,
+            [OpCode.RIGHT] = 1 << 11,
+            [OpCode.INVERT] = 1 << 2,
+            [OpCode.AND] = 1 << 3,
+            [OpCode.OR] = 1 << 3,
+            [OpCode.XOR] = 1 << 3,
+            [OpCode.EQUAL] = 1 << 5,
+            [OpCode.NOTEQUAL] = 1 << 5,
+            [OpCode.SIGN] = 1 << 2,
+            [OpCode.ABS] = 1 << 2,
+            [OpCode.NEGATE] = 1 << 2,
+            [OpCode.INC] = 1 << 2,
+            [OpCode.DEC] = 1 << 2,
+            [OpCode.ADD] = 1 << 3,
+            [OpCode.SUB] = 1 << 3,
+            [OpCode.MUL] = 1 << 3,
+            [OpCode.DIV] = 1 << 3,
+            [OpCode.MOD] = 1 << 3,
+            [OpCode.SHL] = 1 << 3,
+            [OpCode.SHR] = 1 << 3,
+            [OpCode.NOT] = 1 << 2,
+            [OpCode.BOOLAND] = 1 << 3,
+            [OpCode.BOOLOR] = 1 << 3,
+            [OpCode.NZ] = 1 << 2,
+            [OpCode.NUMEQUAL] = 1 << 3,
+            [OpCode.NUMNOTEQUAL] = 1 << 3,
+            [OpCode.LT] = 1 << 3,
+            [OpCode.LE] = 1 << 3,
+            [OpCode.GT] = 1 << 3,
+            [OpCode.GE] = 1 << 3,
+            [OpCode.MIN] = 1 << 3,
+            [OpCode.MAX] = 1 << 3,
+            [OpCode.WITHIN] = 1 << 3,
+            [OpCode.PACK] = 1 << 9,
+            [OpCode.UNPACK] = 1 << 9,
+            [OpCode.NEWARRAY0] = 1 << 4,
+            [OpCode.NEWARRAY] = 1 << 9,
+            [OpCode.NEWARRAY_T] = 1 << 9,
+            [OpCode.NEWSTRUCT0] = 1 << 4,
+            [OpCode.NEWSTRUCT] = 1 << 9,
+            [OpCode.NEWMAP] = 1 << 3,
+            [OpCode.SIZE] = 1 << 2,
+            [OpCode.HASKEY] = 1 << 13,
+            [OpCode.KEYS] = 1 << 4,
+            [OpCode.VALUES] = 1 << 13,
+            [OpCode.PICKITEM] = 1 << 13,
+            [OpCode.APPEND] = 1 << 13,
+            [OpCode.SETITEM] = 1 << 13,
+            [OpCode.REVERSEITEMS] = 1 << 13,
+            [OpCode.REMOVE] = 1 << 4,
+            [OpCode.CLEARITEMS] = 1 << 4,
+            [OpCode.ISNULL] = 1 << 1,
+            [OpCode.ISTYPE] = 1 << 1,
+            [OpCode.CONVERT] = 1 << 11,
```

Comment:
And this one.

Reply:
agree

Reply:
The main bottleneck is the conversion between `Array` and `Struct`. You may have to copy all `Struct` and its children.